msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/AbstractMsalUiTest.java

```diff
@@ -85,35 +86,46 @@
     public final TestRule resetAutomaticTimeRule = new ResetAutomaticTimeZoneTestRule();
 
     @Rule(order = 3)
-    public final TestRule loadLabUserRule = getLabUserQuery() != null
-            ? new LoadLabUserTestRule(getLabUserQuery())
-            : new LoadLabUserTestRule(getTempUserType());
-
-    @Rule(order = 4)
     public final TestRule removeBrokersRule = new RemoveBrokersBeforeTestRule();
 
-    @Rule(order = 5)
+    @Rule(order = 4)
     public ActivityTestRule<MainActivity> mActivityRule =
             new ActivityTestRule(MainActivity.class);
 
-    @Rule(order = 6)
+    @Rule(order = 5)
     public TestRule deviceEnrollmentFailureRecoveryRule = new DeviceEnrollmentFailureRecoveryRule();
 
     @Before
     public void setup() {
+        loadLabUser();
         mScopes = getScopes();
         mBrowser = getBrowser();
 
         // clear all cookies in the browser
         mBrowser.clear();
 
-        mLoginHint = ((LoadLabUserTestRule) loadLabUserRule).getLabUserUpn();
-
         mContext = ApplicationProvider.getApplicationContext();
         mActivity = mActivityRule.getActivity();
         setupPCA();
     }
 
+    private void loadLabUser() {
+        if (getLabUserQuery() != null) {
+            mLoginHint = LabUserHelper.loadUserForTest(getLabUserQuery());
+        } else if (getTempUserType() != null) {
+            mLoginHint = LabUserHelper.loadTempUser(getTempUserType());
+            try {
+                // temp user takes some time to actually being created even though it may be
+                // returned by the LAB API. Adding a wait here before we proceed with the test.
+                Thread.sleep(TEMP_USER_WAIT_TIME);
```

Comment:
Should we bake this into loadTempUser?

Reply:
I'm inclined not to do that for now, and instead let the caller decide if they want to sleep or not. Because let's say if someone called it but wasn't going to use the temp account right away and instead used it some time later (depending on how they have their test), they may not want/need the sleep.